<a href="https://colab.research.google.com/github/shubham14yadav/Marketing_Hypothesis_testing/blob/main/Snapchat_hypothesis_testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np

# Generate dates for the data
dates = pd.date_range(start="2023-01-01", periods=30, freq='D')

# Generate random data for each metric, building each based on the previous one
impressions = np.random.randint(1000, 5000, size=30)
clicks = np.round(impressions * np.random.uniform(0.05, 0.2, size=30)).astype(int)
sign_ups = np.round(clicks * np.random.uniform(0.1, 0.25, size=30)).astype(int)
installs = np.round(sign_ups * np.random.uniform(0.8, 0.9, size=30)).astype(int)
purchases = np.round(installs * np.random.uniform(0.1, 0.5, size=30)).astype(int)
revenue = purchases * np.random.uniform(10, 100, size=30).astype(int)

# Assemble all the data into a DataFrame
data = {
    "Date": dates,
    "Impressions": impressions,
    "Clicks": clicks,
    "Sign Ups": sign_ups,
    "Installs": installs,
    "Purchases": purchases,
    "Revenue": revenue
}
df = pd.DataFrame(data)

# Display the first few rows of the DataFrame
df.head()


Date  Impressions  Clicks  Sign Ups  Installs  Purchases  Revenue
0 2023-01-01         3022     407        60        49         24      528
1 2023-01-02         2187     178        34        30         14      546
2 2023-01-03         3957     484        50        41         10      610
3 2023-01-04         3952     599       149       130         33     1452
4 2023-01-05         4483     631       151       132         13      559

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

# Generate dates and data
dates = pd.date_range(start="2023-01-01", periods=30, freq='D')
impressions = np.random.randint(1000, 5000, size=30)
clicks = np.round(impressions * np.random.uniform(0.05, 0.2, size=30)).astype(int)
sign_ups = np.round(clicks * np.random.uniform(0.1, 0.25, size=30)).astype(int)
installs = np.round(sign_ups * np.random.uniform(0.8, 0.9, size=30)).astype(int)
purchases = np.round(installs * np.random.uniform(0.1, 0.5, size=30)).astype(int)
revenue = purchases * np.random.uniform(10, 100, size=30).astype(int)

# Create DataFrame
df_corrected = pd.DataFrame({
    "Date": dates,
    "Impressions": impressions,
    "Clicks": clicks,
    "Sign Ups": sign_ups,
    "Installs": installs,
    "Purchases": purchases,
    "Revenue": revenue
})

# Add a column for even or odd day
df_corrected['Day Type'] = df_corrected['Date'].dt.day % 2 == 0

# Separate the clicks data into even and odd days
even_clicks = df_corrected[df_corrected['Day Type'] == True]['Clicks']
odd_clicks = df_corrected[df_corrected['Day Type'] == False]['Clicks']

# Perform a t-test
t_stat, p_value = stats.ttest_ind(even_clicks, odd_clicks, equal_var=False)

t_stat, p_value


(0.2527560039491986, 0.8023430692423602)

Let's assume we want to compare the number of clicks between two different segments. For simplicity, we could simulate these segments based on whether the day is even or odd (a simplistic simulation, just for demonstration). This will allow us to compare the mean number of clicks on even days vs. odd days.

Here's how you can perform the T-test using Python:

Split the data into even and odd days based on the date.
Perform a T-test to see if there's a statistically significant difference in the number of clicks between these two groups.

The results of the T-test: a T-statistic of approximately 0.253 and a P-value of about 0.802. Here's what these results mean:

T-statistic (0.253): This value indicates the difference in mean clicks between even and odd days in terms of standard deviations. A value close to 0 suggests that the means of the two groups are very similar.

P-value (0.802): This value tells you the probability of observing a difference as large as (or larger than) what is observed, under the assumption that there is no actual difference between the groups (null hypothesis is true). A P-value much higher than 0.05 (typically used as a threshold for statistical significance) suggests that the observed difference in clicks between even and odd days is not statistically significant.
Given these results, we can conclude that there is no significant difference in the number of clicks on even versus odd days within this dataset. This implies that the day type (even or odd) does not have a statistically significant impact on the number of clicks.

the hypothesis would have been structured as follows:

Null Hypothesis (H0)
H0: There is no difference in the mean number of clicks between even days and odd days.

This implies that the day type (even or odd) has no effect on the number of clicks.

Alternative Hypothesis (H1)
H1: There is a difference in the mean number of clicks between even days and odd days.

This suggests that the day type might influence the number of clicks.
Given the P-value of 0.802, which is significantly higher than the common threshold of 0.05, we fail to reject the null hypothesis. This means the data does not provide strong evidence to support the alternative hypothesis that even and odd days affect the number of clicks differently.

In [ ]:
df_corrected

Date  Impressions  Clicks  Sign Ups  Installs  Purchases  Revenue  \
0  2023-01-01         1954     274        55        48         19      456   
1  2023-01-02         4760     422        89        77         18      396   
2  2023-01-03         2813     386        71        59         22     2002   
3  2023-01-04         4810     665       110        88         16      976   
4  2023-01-05         2072     151        37        32          3      261   
5  2023-01-06         1872     190        23        20          8      736   
6  2023-01-07         3784     394        90        78         37     3515   
7  2023-01-08         1840     258        46        41         12      996   
8  2023-01-09         3159     631       130       108         13      858   
9  2023-01-10         2619     231        53        43         21     1827   
10 2023-01-11         1623     229        25        21          3      165   
11 2023-01-12         1781     127        19        17          3      288   
12 2023-01-13         4594     337        65        57          7      630   
13 2023-01-14         3963     287        62        53         22     1958   
14 2023-01-15         3553     707       121        98         20      820   
15 2023-01-16         2625     175        30        26         12      216   
16 2023-01-17         1323     217        32        27          6      510   
17 2023-01-18         4859     804       176       148         42     1764   
18 2023-01-19         3762     464        93        76         35     2100   
19 2023-01-20         1749     349        85        70         35      490   
20 2023-01-21         3718     315        58        50          9      216   
21 2023-01-22         1832     132        16        13          2       86   
22 2023-01-23         1421     267        56        49         21      693   
23 2023-01-24         2662     415        62        55          6      378   
24 2023-01-25         2190     208        40        34          7      273   
25 2023-01-26         4839     420        91        76         13      663   
26 2023-01-27         1977     140        17        15          4      348   
27 2023-01-28         4418     361        57        46         21      693   
28 2023-01-29         3378     278        50        41         18      774   
29 2023-01-30         3054     407        81        70          9      540   

    Day Type  
0      False  
1       True  
2      False  
3       True  
4      False  
5       True  
6      False  
7       True  
8      False  
9       True  
10     False  
11      True  
12     False  
13      True  
14     False  
15      True  
16     False  
17      True  
18     False  
19      True  
20     False  
21      True  
22     False  
23      True  
24     False  
25      True  
26     False  
27      True  
28     False  
29      True